src/supermarket/app/controllers/cookbooks_controller.rb

```diff
@@ -308,4 +309,17 @@ def apply_filters
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
+
+  def classify_metrics(metric_results)
```

Comment:
I'm pondering some database tricks we can use to refine this classification of results and preloading of results and the quality_metric with which they're associated. Judicious application of `.includes(:quality_metrics)` and some `.group_by`ing.